force-app/main/default/classes/OpportunityUtilities.cls

```diff
@@ -0,0 +1,259 @@
+            if (opp.Type == null) {
+                opp.Type = 'New Customer';
+            }
+        }
+
+    }
+
+
+    // Create a new Task for newly inserted Opportunities
+    public static void newOppyGetsTask(List<Opportunity> oppys){
+
+        //Creating a list to hold the tasks
+        List<Task> tasks = new List<Task>();
+
+        for (Opportunity opp : oppys) {
+            Task tsk = new Task();
+                tsk.Subject = 'Call Primary Contact';
+                tsk.WhatId = opp.Id;
+                tsk.WhoId = opp.Primary_Contact__c;
+                tsk.OwnerId = opp.OwnerId;
+                tsk.ActivityDate = Date.today().addDays(3);
+
+                //adding each task to the above list
+                tasks.add(tsk);
+            }
+        insert tasks;
+
+    }
+
+    // Append Stage changes in Opportunity Description
+    public static void appendStageChangeInOppyDescription(List<Opportunity> oppys) {
+
+        // Check if the stage change has already been handled in this transaction
+        if (OpportunityUtilities.isStageChangeHandled) {
+            return; // Exit if already handled to prevent recursion
+        }
+    
+        // Set the flag to true to prevent further updates in the same transaction
+        OpportunityUtilities.isStageChangeHandled = true;
+    
+        system.debug(oppys);
+
+        for (Opportunity opp : oppys) {
+            if (opp.StageName != null) {
+                opp.Description += 'Stage Change:' + opp.StageName;
+            }
+        }
+    }
+
+
+    // Trigger should only fire on update.
+
+    // When an opportunity is updated set the primary contact on the opportunity to the contact with the title of 'CEO'
+    public static void updatePrimaryContact(List<Opportunity> oppys){
+
+        List<Id> accountIds = new List<Id>();
+
+            for (Opportunity opp : oppys) {
+                if (opp.AccountId != null) {
+                    accountIds.add(opp.AccountId);
+                }
+            }
+
+            Map<Id, Contact> accountCEOs = new Map<Id, Contact>([SELECT Id, AccountId, Title FROM Contact WHERE AccountId IN :accountIds AND Title = 'CEO']);
+
+            Map<Id, Contact> accountIdContactMap = new Map<Id, Contact>();
+
+                for (Contact con : accountCEOs.values()) {
+                    accountIdContactMap.put(con.AccountId, con);
+                }
+
+                for(Opportunity opp : oppys){
+                    if(opp.Primary_Contact__c == null && accountIdContactMap.containsKey(opp.AccountId)){
+                        opp.Primary_Contact__c = accountIdContactMap.get(opp.AccountId).Id;
+                    }
+                }
+    }    
+
+
+    // Assigns a primary contact with the title of 'VP Sales' to undeleted Opportunities 
+    // Only updates the Opportunities that don't already have a primary contact   
+
+    public static void assignPrimaryContact(Map<Id, Opportunity> oppNewMap) {
+
```

Comment:
updatePrimaryContact and assignPrimaryContact are basically variations of the same thing. This is the sort of area you want to look for ways to combine repeated code. Figure out which part is slightly different and then account for that in the code and pass in something that specifies that difference.